Rakefile

```diff
@@ -1,72 +1,50 @@
-require "json"
 require "pathname"
 
-CONFIG_FILE  = Pathname.new(__dir__).join(ENV["CNFFILE"] || "config.json")
 BUILD_ROOT   = Pathname.new(ENV["OBJDIR"] || "build")
 INSTALL_ROOT = Pathname.new(ENV["PREFIX"] || "/opt/ood")
+VENDOR_DIR   = Pathname.new(ENV["VENDOR_DIR"] || BUILD_ROOT.join("vendor").expand_path )
 
-def all_components
-  JSON.parse(CONFIG_FILE.read).map { |c| Component.new(c) }
-end
-
-class Component
-  attr_reader :name
-  attr_reader :url
-  attr_reader :tag
-
-  def initialize(opts = {})
-    @name = opts.fetch("name") { raise ArgumentError, "No name specified. Missing argument: name" }.to_s
-    @url  = opts.fetch("url")  { raise ArgumentError, "No url specified. Missing argument: url" }.to_s
-    @tag  = opts.fetch("tag")  { raise ArgumentError, "No tag specified. Missing argument: tag" }.to_s
-    @app  = opts.fetch("app", nil)
-  end
-
-  def app?
-    !!@app
-  end
-
-  def build_root
-    app? ? BUILD_ROOT.join("apps", name) : BUILD_ROOT.join(name)
-  end
-
-  def download_url
-    "#{url}/archive/#{tag}.tar.gz"
-  end
+task :default => :build
 
-  def file_url
-    "apps/#{name}/."
-  end
+def apps
+  %w(activejobs bc_desktop dashboard file-editor files myjobs shell).map {|name| BUILD_ROOT.join("apps", name) }
 end
 
-task :default => :build
+def build_apps
+  app_build_dir = BUILD_ROOT.join("apps")
+  rm_rf app_build_dir if app_build_dir.directory?
+  mkdir_p BUILD_ROOT unless BUILD_ROOT.directory?
+  cp_r "apps", BUILD_ROOT
```

Comment:
As discussed, we don't need a separate BUILD_ROOT anymore. "building the apps" should just cd into the app directory and install the dependencies.

Reply:
Of course now I realize that the spec file was using the build directory to find the apps so this change might require an update to the spec file before RPM building works.